up.ts

```diff
@@ -109,12 +136,17 @@ const start = async function () {
         stdout: 'piped',
         stderr: 'piped'
     });
+
+    const hardhatReadyEvent = 'hardhat ready';
     // NOTE: the process should keep running until we kill it
-    pipeNamedSubprocess(cyan('Hardhat'), hardhat.stdout, Deno.stdout);
+    pipeNamedSubprocess(cyan('Hardhat'), hardhat.stdout, Deno.stdout, {
+        readyEvent: hardhatReadyEvent,
+        message: 'Mined empty block'
+    });
     pipeNamedSubprocess(red('Hardhat'), hardhat.stderr, Deno.stderr);
 
-    // very naive way to let the Hardhat node start before deploying to it
-    await delay(31 * 1000);
```

Comment:
Switching from the naive "just wait for a while", to listening to the process for an indication it's ready.